pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/TopicTransactionBuffer.java

```diff
@@ -129,6 +131,23 @@ public TopicTransactionBuffer(PersistentTopic topic) {
         this.recover();
     }
 
+    @VisibleForTesting
+    TopicTransactionBuffer(PersistentTopic topic, AbortedTxnProcessor snapshotAbortedTxnProcessor,
+                           AbortedTxnProcessor.SnapshotType snapshotType) {
+        super(State.None);
+        this.topic = topic;
+        this.timer = topic.getBrokerService().getPulsar().getTransactionTimer();
+        this.takeSnapshotIntervalNumber = topic.getBrokerService().getPulsar()
+                .getConfiguration().getTransactionBufferSnapshotMaxTransactionCount();
+        this.takeSnapshotIntervalTime = topic.getBrokerService().getPulsar()
+                .getConfiguration().getTransactionBufferSnapshotMinTimeInMillis();
+        this.maxReadPosition = topic.getManagedLedger().getLastConfirmedEntry();
+        this.snapshotAbortedTxnProcessor = snapshotAbortedTxnProcessor;
+        this.snapshotType = snapshotType;
+        this.maxReadPositionCallBack = topic.getMaxReadPositionCallBack();
+        this.recover();
+    }
```

Comment:
This new constructor overload has much duplicated code with the original constructor. Please reuse the code. It  only just reduces duplicated code, but also simulate the real case more likely.

Here is a bad example. Assuming there is a `Demo` class whose constructor is:

```java
public Demo(Argument arg) {
    this.field1 = arg.f();
    this.field2 = arg.g();
}
```

If you wanted to pass a mocked `field2` and added a new constructor;

```java
Demo(Argument arg, Field2 field2) {
    this.field1 = arg.h();
    this.field2 = field2;
}
```

Then the new constructor will be meaningless. Because now `field` is not `arg.f()` anymore.